plugins/hydra_ax_sweeper/tests/test_ax_sweeper_plugin.py

```diff
@@ -189,21 +192,57 @@ def test_configuration_set_via_cmd_and_default_config(hydra_sweep_runner) -> Non
         assert "quadratic_y" in best_parameters
 
 
-def test_ax_logging(tmpdir: Path) -> None:
+@pytest.mark.parametrize(
+    "cmd_arg, expected_str",
+    [
+        (
+            "polynomial.y=choice(-1, 0, 1)",
+            "polynomial.y: choice=[-1, 0, 1]",
+        ),
+        (
+            "polynomial.y=range(-1, 2)",
+            "polynomial.y: choice=[-1, 0, 1]",
+        ),
+        (
+            "polynomial.y=range(-1, 3, 1)",
+            "polynomial.y: choice=[-1, 0, 1, 2]",
+        ),
+        (
+            "polynomial.y=range(-1, 2, 0.5)",
+            "polynomial.y: choice=[-1.0, -0.5, 0.0, 0.5, 1.0, 1.5]",
+        ),
+        (
+            "polynomial.y=int(interval(-1, 2))",
+            "polynomial.y: range=[-1, 2]",
+        ),
+        (
+            "polynomial.y=interval(-1, 2)",
+            "polynomial.y: range=[-1.0, 2.0]",
+        ),
+        (
+            "polynomial.y=2",
+            "polynomial.y: fixed=2",
+        ),
+        (
+            "polynomial.y=2.0",
+            "polynomial.y: fixed=2.0",
+        ),
+    ],
+)
+def test_ax_logging(tmpdir: Path, cmd_arg: str, expected_str: str) -> None:
     cmd = [
         sys.executable,
         "tests/apps/polynomial.py",
         "-m",
         "hydra.run.dir=" + str(tmpdir),
-        "polynomial.x=-5:-2",
-        "polynomial.y=-1,1",
+        "polynomial.x=interval(-5, -2)",
         "polynomial.z=10",
         "hydra.sweeper.ax_config.max_trials=2",
-    ]
+    ] + [cmd_arg]
     result = subprocess.check_output(cmd).decode("utf-8").rstrip()
```

Comment:
why is this using a subprocess instead of testing the function directly?

Reply:
We are reading the output written to the console (not saved)